build_system/src/build.rs

```diff
@@ -29,6 +29,16 @@ impl BuildArg {
                         );
                     }
                 }
+                "--sysroot" => {
+                    if let Some(arg) = args.next() {
+                        build_arg.flags.push("--sysroot".to_string());
```

Comment:
This is not what should be done.

Instead, this new `--sysroot` flag that you added should be used to determine if we should build the sysroot or not. This happens [here](https://github.com/rust-lang/rustc_codegen_gcc/blob/master/build_system/src/build.rs#L223).

Reply:
instead we should call `build_sysroot(&env, &args.config_info)?;` on the if check

Reply:
In this function, set a new field that you can add on this struct to `true` and in the link from my comment above, wrap the call to `build_sysroot` in an `if` that will check if this new field is set to `true`.

Reply:
adding a boolean flag on this `BuildArg`, then set it to TRUE on `--sysroot` passed